### PR TITLE
Long term stats, cumulative energy & manufacturer info

### PR DESCRIPTION
**Pull request by khcnz**
Implement long term stats changes to sensor class
**Draft** changes to support a cumulative power sensor (awaiting pychonet update)
Update device name to match the user's configured name in config flow
Prefix entity names to include user's configured name to prevent conflicts in names and ids when multiple devices of the same class exist

### CODE REVIEW

File: custom_components/echonetlite/__init__.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def __init__(self, instance, api, entry):
                 self._user_options[ENL_SWING_MODE] = entry.options.get("swing_mode")
 
         self._uid = self._api._state[self._host]["instances"][self._eojgc][self._eojcc][self._eojci][ENL_UID]
-        if self._uid == None:
+        if self._uid is None:
```
Comment (khcnz) [inline custom_components/echonetlite/__init__.py:184]: I understand is you should always use "is None" in python

File: custom_components/echonetlite/sensor.py [hunk 2]
```diff
@@ -26,35 +33,46 @@ async def async_setup_entry(hass, config, async_add_entities, discovery_info=Non
...
-                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], hass.config.units, config.title))
+                    entities.append(EchonetSensor(entity['echonetlite'], op_code, ENL_SENSOR_OP_CODES['default'], config.title))
     async_add_entities(entities, True)
 
 
-class EchonetSensor(Entity):
+class EchonetSensor(SensorEntity):
     """Representation of an ECHONETLite Temperature Sensor."""
 
-    def __init__(self, instance, op_code, attributes, units: UnitSystem, name=None) -> None:
```
Comment (khcnz) [inline custom_components/echonetlite/sensor.py:46]: I've removed units from this sensor as echonet is always Celsius AFAIK so no need for this to be a parameter

File: custom_components/echonetlite/climate.py [hunk 3]
```diff
@@ -108,7 +109,7 @@ def device_info(self):
             "identifiers": {
                   (DOMAIN, self._connector._uid, self._connector._instance._eojgc, self._connector._instance._eojcc, self._connector._instance._eojci)
             },
-            "name": EOJX_CLASS[self._connector._instance._eojgc][self._connector._instance._eojcc]
+            "name": self._device_name
```
Comment (khcnz) [inline custom_components/echonetlite/climate.py:112]: This stops issues with multiple devices all being called HomeAirConditioner which was a) confusing and b) causing issues when adding more than one item.

File: custom_components/echonetlite/const.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
 """Constants for the echonetlite integration."""
-from homeassistant.const import CONF_ICON, CONF_NAME, CONF_TYPE
+from homeassistant.const import CONF_ICON, CONF_NAME, CONF_TYPE, DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_ENERGY
+from homeassistant.components.sensor import ATTR_STATE_CLASS, STATE_CLASS_MEASUREMENT, STATE_CLASS_TOTAL_INCREASING
 from pychonet.HomeAirConditioner import ENL_FANSPEED, ENL_AIR_VERT, ENL_AIR_HORZ, ENL_AUTO_DIRECTION, ENL_SWING_MODE, FAN_SPEED, AIRFLOW_VERT, AIRFLOW_HORIZ, AUTO_DIRECTION, SWING_MODE
 
 DOMAIN = "echonetlite"
-
-SENSOR_TYPE_TEMPERATURE = "temperature"
```
Comment (khcnz) [inline custom_components/echonetlite/const.py:7]: Reused inbuilt HA constant for a temperature sensor device class DEVICE_CLASS_TEMPERATURE

**Review by khcnz — COMMENTED** [review]
Some notes to help with a review

File: custom_components/echonetlite/sensor.py (hunk 2 above)
Comment (scottyphillips) [reply custom_components/echonetlite/sensor.py:46]: The sensor can be used by lots of things not just temperature values so I don’t really think this change is a good idea

Comment (khcnz) [reply custom_components/echonetlite/sensor.py:46]: I saw a comment from you about this but it's disappeared.  The idea with the native values thing is to never do conversions based on the users configured unit systems - always pass the raw values the system gives you and let HA do any conversions within the sensor platform.

**Comment by scottyphillips** [comment]
Yeah fair enough I delete my original comment because I figured there was more to that change.

Scott

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by khcnz** [comment]
image
image

**Comment by khcnz** [comment]
Long term stats
image

**Review by scottyphillips — APPROVED** [review]

**Comment by scottyphillips** [comment]
I am actually deeply envious that you can measure your HVAC power consumption. I feel robbed...

**Pull request merged**